deploy/util/targetprojectutils.py

```diff
@@ -0,0 +1,14 @@
+def extra_target_project_make_args(targetproject: str, deploydir: str) -> str:
+    """Create extra make args for target projects that live outside FireSim.
+    Eventually will be replaced by using YAML hooks.
+
+    Args:
+        targetproject: make arg corresponding to TARGET_PROJECT.
+        deploydir: Deploy directory of FireSim
+
+    Returns:
+        any extra args needed for FireSim's make command to function
+    """
+    if targetproject in ["firesim", "bridges"]:
```

Comment:
So just to check my understanding...
If you run manager commands (e.g. `firesim infrasetup`), it calls into the makefiles inside chipyard to generate the target design + bridges to generate the FIRRTL & annotations, and then you call midas inside FireSim?

Reply:
`firesim infrasetup` calls the `firesim/sim/Makefile`. To build RTL for Chipyard, it needs to have a reference to the makefrags in Chipyard. Those makefrags tell the FireSim `Makefile` what to do to get `.fir/.anno.json` and how to use the bridges it has. Those makefrags call the Chipyard `Makefile` directly to get the FIRRTL. So the order is:

`firesim/sim/Makefile` -> Chipyard MakeFrags -> Chipyard `Makefile` -> Got FIRRTL/ANNOs here -> Back to `firesim/sim/Makefile` to finish GG compilation.

Reply:
Yikes